### PR TITLE
Clone in Parallel

### PR DESCRIPTION
**Pull request by instanceof**
Update to process each submodule in parallel and collect the submodule output in a log buffer.
Colorize the output on error.

Needs a major/minor version change.

@bgrainger, @jacobcarpenter

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -90,17 +93,20 @@ function createSolutionInfos() {
...
-  const [owner, repo] = next.split('/');
+  const [owner, repo] = name.split('/');
   const submodule = {
+    _logs: [],
     owner,
     repo,
     branch,
-    remoteUrl: `git@git:${owner}/${repo}.git`
+    remoteUrl: `git@git:${owner}/${repo}.git`,
+    log(message, indent=0) { this._logs = this._logs.concat(message.split(/\r?\n/).map((line) => { return Array(indent + 1).join(' ') + line; })); },
```
Comment (jacobcarpenter) [inline src/index.js]: [String.prototype.repeat()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/repeat)

As in

``` js
line => ' '.repeat(indent + 1) + line
```

Comment (instanceof) [inline src/index.js]: That is only available in ES6 and if you try it in a node repl it is not defined.

Also, for indent=0, you would still get one space.

With Array(0 + 1).join(' ') you get an empty string.  For indent=2, you get an array with 3 elements which are joined to give two spaces.

Comment (jacobcarpenter) [inline src/index.js]: `clone-leeroy` _is_ ES6. If you need an ES6 repl, use `babel-node`.

Sorry I missed the point of the + 1 in the old code. `' '.repeat(indent) + line` should work instead, right?

Comment (instanceof) [inline src/index.js]: Yes, I tried it with babel-node and also built the project with gulp and it produces ' '.repeat(indent) + line, but when you try to run it in node v0.12.7, you get `undefined is not a function`.

I think we have to wait for a future version of node or run node with `--harmony`.

Changing the first line of dist/index.js to:

```
#!/usr/bin/env node --harmony
```

allows repeat() to work.  I will figure out how to get gulp to add the flag.

**Pull request merged**